docs/API.md

```diff
@@ -113,7 +113,7 @@ All error states that occur after the RPC method has been identified are reporte
 This API provides three classes of methods:
 
 * *Unrestricted* methods are available to anyone with access to the API.
-* *Authenticated* methods have parameters for *username* and *api key*
+* *Authenticated* methods have parameters for [*username*][Username] and [*api key*][API key]
```

Comment:
Have you considered putting the asterisks outside the link text, i.e. `*[username]*`? Maybe even skipping the asterisks entirely? Anyway the matching of link labels is case-insensitive.

Reply:
The asterisks inside or outside the link text should be the same, but I think we should have the defined types, such as *username*, in italics to distinguish them from other links.

Reply:
> Have you considered putting the asterisks outside the link text, i.e. *[username]*?

I think I have, and I could not come to a good solution. Any good practice in Markdown on nesting styles?

> Maybe even skipping the asterisks entirely?

I recently rediscovered a document I read a long time ago: https://github.com/zonemaster/zonemaster-backend/blob/master/docs/TypographicConventions.md which says:

> Italic text is used for [...] data types defined in the API document

I don't know if we are consistent with other conventions from this document but I think we can keep the italic here.

Reply:
The following will work:
```
*[username][Username]*
```
I.e.:

*[username][Username]*

[Username]: https://github.com/zonemaster/zonemaster-backend/blob/987812241a41916b9c2b9f30dd8f882d98b4365a/docs/API.md#username

Reply:
I was working on this and I realized that there is a case were we can't swap formatting and links:

```
[`processed as a link`][link]
`[processed as a code block][link]`

[link]: https://example.com
```

[`processed as a link`][link]
`[processed as a code block][link]`

[link]: https://example.com

For consistency, I'm chosing to go with the formatting within the link brackets:

```
[*italic*][link]
[`code`][link]
[**bold**][link]
```

Reply:
Fine for me. One should avoid '`' and '*' in the link list, and you have done that.